cmd/create/machinepool/helper.go

```diff
@@ -65,6 +67,25 @@ func getSubnetFromUser(cmd *cobra.Command, r *rosa.Runtime, isSubnetSet bool,
 
 // getSubnetOptions gets one of the cluster subnets and returns a slice of formatted VPC's private subnets.
 func getSubnetOptions(r *rosa.Runtime, cluster *cmv1.Cluster) ([]string, error) {
+
+	lowestLocalZoneSupportVer, err := ver.NewVersion(ocm.LowestLocalZoneSupport)
+	if err != nil {
+		return nil, err
+	}
+
+	checkLocalZone := false
+	if version, ok := cluster.GetVersion(); ok {
+		if rawID, ok := version.GetRawID(); ok {
+			clusterVer, err := ver.NewVersion(rawID)
+			if err != nil {
+				return nil, err
+			}
+			if !clusterVer.GreaterThanOrEqual(lowestLocalZoneSupportVer) {
+				checkLocalZone = true
+			}
+		}
+	}
```

Comment:
Would it be possible if we pull this out into its own function. It would help flatten out the nesting we have as we could return early if `!ok` or `err`

Reply:
Could we add a unit test for this also please :pray:

Reply:
@ciaranRoche 
new changes committed, thanks for the comments.